autosetup.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# global variables
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+RESET_PRINT='\033[0;0m'
+
+installed_packages="0"
+required_packages="0"
+venv_name="venv_sapsam"
+kernel_name="sapsam_kernel"
+
+# checks for pip
+required_packages=$((required_packages + 1))
+if pip --version &>/dev/null; then
+    echo -e "\xE2\x9C\x94 pip installed..."
+    installed_packages=$((installed_packages + 1))
+else
+    echo "pip not installed"
+    echo "try installing pip with for example (on macOS): brew install pip"
+    echo "exiting setup script..."
+    exit 1
+fi
+
+# checks for python3
+required_packages=$((required_packages + 1))
+if command -v python3 &>/dev/null; then
+    echo -e "\xE2\x9C\x94 python3 installed..."
+    installed_packages=$((installed_packages + 1))
+else
+    echo "python3 not installed"
+    echo "try installing python3 with for example (on macOS): brew install python3"
+    echo "exiting setup script..."
+    exit 1
+fi
+
+# checks for venv
+required_packages=$((required_packages + 1))
+if python3 -m venv --help &>/dev/null; then
+    echo -e "\xE2\x9C\x94 venv installed..."
+    installed_packages=$((installed_packages + 1))
+else
+    echo "venv module not installed"
+    echo "exiting setup script..."
+    exit 1
+fi
+
+# checks for jupyter
+required_packages=$((required_packages + 1))
+if pip list | grep jupyter &>/dev/null; then
+    echo -e "\xE2\x9C\x94 jupyter installed..."
+    installed_packages=$((installed_packages + 1))
+else
+    echo "jupyter module not installed"
+    echo "try installing jupyter with for example: pip install jupyter notebook"
+    echo "exiting setup script..."
+    exit 1
+fi
+
+# checks for notebook
+required_packages=$((required_packages + 1))
+if pip list | grep notebook &>/dev/null; then
+    echo -e "\xE2\x9C\x94 notebook installed..."
+    installed_packages=$((installed_packages + 1))
+else
+    echo "notebook module not installed"
+    echo "exiting setup script..."
+    exit 1
+fi
+
+echo -e "${GREEN}$installed_packages out of $required_packages pre-required packages installed${RESET_PRINT}"
+
+# creates the virtual environment
+if [ -d "$venv_name" ]; then
+    echo "virtual environment '$venv_name' already exists"
+else
+    python3.8 -m venv "$venv_name"
```

Comment:
Why exactly `3.8`?

Reply:
You are right, don't really know why I put the .8 here :D